datafusion/functions-aggregate/src/covariance.rs

```diff
@@ -93,18 +95,29 @@ impl AggregateUDFImpl for CovarianceSample {
         Ok(DataType::Float64)
     }
 
-    fn state_fields(
-        &self,
-        name: &str,
-        _value_type: DataType,
-        _ordering_fields: Vec<Field>,
-    ) -> Result<Vec<Field>> {
+    fn field(&self, args: datafusion_expr::function::FieldArgs) -> Result<Field> {
+        Ok(Field::new(args.name, args.return_type.clone(), true))
+    }
+
+    fn state_fields(&self, args: StateFieldsArgs) -> Result<Vec<Field>> {
         Ok(vec![
-            Field::new(format_state_name(name, "count"), DataType::UInt64, true),
-            Field::new(format_state_name(name, "mean1"), DataType::Float64, true),
-            Field::new(format_state_name(name, "mean2"), DataType::Float64, true),
             Field::new(
-                format_state_name(name, "algo_const"),
+                format_state_name(args.name, "count"),
+                DataType::UInt64,
+                args.nullable,
+            ),
+            Field::new(
+                format_state_name(args.name, "mean1"),
+                DataType::Float64,
+                args.nullable,
```

Comment:
aren't some of these fields nullable even if their input is not nullable (e.g. mean with no inputs is null)?

Reply:
yes, this should be `true`